pkg/foundation/semaphore/semaphore.go

```diff
@@ -16,71 +16,102 @@ package semaphore
 
 import (
 	"sync"
-
-	"github.com/conduitio/conduit/pkg/foundation/cerrors"
 )
 
-// Simple provides a way to bound concurrent access to a resource. It only
-// allows one caller to gain access at a time.
+// Simple provides a way to lock concurrent access to a resource. It only allows
+// one caller to gain access at a time.
 type Simple struct {
```

Comment:
It would be good to have an example of a semaphore's usage here: you create one, enqueue a ticket, acquire, then release.

Reply:
I added a simple example with 2 tickets.